core/src/impl/Kokkos_AnalyzePolicy.hpp

```diff
@@ -49,28 +49,65 @@
 #include <Kokkos_Concepts.hpp>
 #include <impl/Kokkos_Tags.hpp>
 #include <impl/Kokkos_GraphImpl_fwd.hpp>
+#include <impl/Kokkos_Error.hpp>
+#include <type_traits>
+
 namespace Kokkos {
+namespace Experimental {
+struct DesiredOccupancy {
+  int _occ = 100;
+  explicit DesiredOccupancy(int occ) : _occ(occ) {
+    KOKKOS_EXPECTS(0 <= occ && occ <= 100);
+  }
+  explicit operator int() const { return _occ; }
+  DesiredOccupancy() = default;
+};
+struct DefaultBlockSizeDeduction {};
+namespace Impl {
+template <typename>
+struct controls_block_size_deduction : std::false_type {};
+template <>
+struct controls_block_size_deduction<DesiredOccupancy> : std::true_type {};
+template <>
+struct controls_block_size_deduction<DefaultBlockSizeDeduction>
+    : std::true_type {};
+}  // namespace Impl
+}  // namespace Experimental
+
 namespace Impl {
 
 template <typename ExecutionSpace = void, typename Schedule = void,
           typename WorkTag = void, typename IndexType = void,
           typename IterationPattern = void, typename LaunchBounds = void,
           typename MyWorkItemProperty =
               Kokkos::Experimental::WorkItemProperty::None_t,
-          typename IsGraphKernel = std::false_type>
+          typename IsGraphKernel      = std::false_type,
+          typename BlockSizeDeduction = Experimental::DefaultBlockSizeDeduction>
 struct PolicyTraitsBase {
   using type = PolicyTraitsBase<ExecutionSpace, Schedule, WorkTag, IndexType,
                                 IterationPattern, LaunchBounds,
-                                MyWorkItemProperty, IsGraphKernel>;
-
-  using execution_space    = ExecutionSpace;
-  using schedule_type      = Schedule;
-  using work_tag           = WorkTag;
-  using index_type         = IndexType;
-  using iteration_pattern  = IterationPattern;
-  using launch_bounds      = LaunchBounds;
-  using work_item_property = MyWorkItemProperty;
-  using is_graph_kernel    = IsGraphKernel;
+                                MyWorkItemProperty, BlockSizeDeduction>;
```

Comment:
Rebase gone wrong here.
```suggestion
                                MyWorkItemProperty, IsGraphKernel, BlockSizeDeduction>;
```